pkg/odo/cli/logs/logs.go

```diff
@@ -99,17 +140,18 @@ func (o *LogsOptions) Run(ctx context.Context) error {
 			}
 		}
 	}
+
 	return nil
 }
 
 func getUniqueContainerName(name string, uniqueNames map[string]struct{}) string {
 	if _, ok := uniqueNames[name]; ok {
 		// name already present in uniqueNames; find another name
-		// first check if last character in name is a number; if so increment it, else append name with 1
-		last, err := strconv.Atoi(string(name[len(name)-1]))
+		// first check if last character in name is a number; if so increment it, else append name with [1]
+		last, err := strconv.Atoi(string(name[len(name)-2]))
```

Comment:
I would create an abstraction for this, to make the character manipulations elsewhere.

Also, I would split in the first '[' and right-trim the ']' from the last part to get the number, to be sure to support multiple-digit numbers